src/index.js

```diff
@@ -0,0 +1,115 @@
+const request = require('request')
+const fixWkt = require('./fixwkt')
+const esriCodes = require('@esri/proj-codes')
+const esriCodesDeprecated = require('esri-proj-codes')
+
+/**
+ * Define the SpatialReference class
+ */
+class SpatialReference {
+  constructor (options) {
+    // If a db was passed in, ensure it has required methods, then add to instance
+    if (options.db && options.db.getWKT && options.db.insertWKT) {
+      this.db = options.db
+    } else if (options.db) {
+      throw new Error('Passed in db does not support getWKT or insertWKT')
+    }
+    // Add optional logger to instance
+    this.logger = options.logger
+  }
+
+  /**
+   * Convert a wkid to a wkt, by 1) looking in db, 2) check ESRI lookups, 3) asking epsg.io
+   * @param {integer} wkid
+   * @param {function} callback
+   */
+  async wkidToWkt (wkid, callback) {
+    try {
+      let wkt
+      // is a db used? check there first
+      if (this.db) {
+        wkt = await getFromDB.call(this, wkid)
+        if (wkt) return callback(null, fixWkt(wkt))
+      }
+      // No db, or not found in DB
+      if (!wkt) wkt = getWktFromEsri(wkid) || await getFromApi(wkid)
+
+      // Insert into DB
+      if (this.db) {
+        return this.db.insertWKT(wkid, wkt, (err) => {
+          if (err) log.call(this, 'error', 'Failed to insert WKT into database: ' + wkid + ' ' + wkt + ', ' + err.message)
+          callback(err, fixWkt(wkt))
+        })
+      }
+      callback(null, fixWkt(wkt))
+    } catch (err) {
+      // log
+      log(this, 'error', err.message)
+      callback(err)
+    }
+  }
+
+  /**
+   * Wrapper around ESRI lookup
```

Comment:
this is pedantic and non-critical to the point of being embarrassing, but when i see "ESRI" (instead of "Esri") i automatically assume the code was written prior to 2010. :smile:

![dude](https://dbouwman.files.wordpress.com/2010/07/jack-1-blog-scaled500.jpg?w=500&h=388)

Reply:
ha, ok.  I guess I missed that in orientation.